tokeniser.go

```diff
@@ -74,49 +97,63 @@ func (t *Tokeniser) endWord() {
 	t.inWord = false
 }
 
-// Tokenise feeds raw bytes into a Tokeniser.
-//
-// If the bytes include the ending of one or more command lines, those lines
-// shall be returned, as a slice of lines represented as slices of
-// word-strings.  Else, the slice shall be empty.
+// Tokenise reads a tokenised line from the Reader.
 //
-// Tokenise may return an error if its current word gets over-full.  In this
-// case, lines will contain the lines it managed to tokenise before keeling
-// over, count will contain the number of bytes processed (including the byte
-// causing the error), and the tokeniser will remain in error until replaced or
-// the current word is ended.
-func (t *Tokeniser) Tokenise(data []byte) (lines [][]string, count uint64, err error) {
-	count = 0
-	for _, b := range data {
-		// Exit early if the current word has become over-full.  This
-		// lets the caller handle the error without silently masking it
-		// if we manage to progress.
-		if t.err != nil {
-			break
-		}
+// Tokenise may return an error if its current word gets over-full, or the Reader chokes.
+// In the former case, the Tokeniser will need to be replaced.
+func (t *Tokeniser) Tokenise() ([]string, error) {
```

Comment:
Generally with this struct is a lot of functions that take nothing and return nothing, only modifying internal state. This is a "code smell" imo and should be avoided. I appreciate that this is effectively a state machine (maybe refactor it into an actual one?) but I think this should be avoided